auth0/resource_auth0_guardian.go

```diff
@@ -384,25 +365,17 @@ func updateGuardian(ctx context.Context, d *schema.ResourceData, m interface{})
 func deleteGuardian(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	api := m.(*management.Management)
 
-	if err := api.Guardian.MultiFactor.Phone.Enable(false); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := api.Guardian.MultiFactor.Email.Enable(false); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := api.Guardian.MultiFactor.OTP.Enable(false); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := api.Guardian.MultiFactor.WebAuthnRoaming.Enable(false); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := api.Guardian.MultiFactor.WebAuthnPlatform.Enable(false); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := api.Guardian.MultiFactor.DUO.Enable(false); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := api.Guardian.MultiFactor.Push.Enable(false); err != nil {
+	result := multierror.Append(
+		api.Guardian.MultiFactor.Phone.Enable(false),
+		api.Guardian.MultiFactor.Email.Enable(false),
+		api.Guardian.MultiFactor.OTP.Enable(false),
+		api.Guardian.MultiFactor.RecoveryCode.Enable(false),
+		api.Guardian.MultiFactor.WebAuthnRoaming.Enable(false),
+		api.Guardian.MultiFactor.WebAuthnPlatform.Enable(false),
+		api.Guardian.MultiFactor.DUO.Enable(false),
+		api.Guardian.MultiFactor.Push.Enable(false),
```

Comment:
It's better if we try al of these updates and report on any accumulated errors instead of failing at the first and not doing any of the others.